src/format.js

```diff
@@ -85,13 +85,9 @@ export function formatTime(config, state, value, options = {}) {
     let defaults        = format && getNamedFormat(formats, 'time', format);
     let filteredOptions = filterProps(options, DATE_TIME_FORMAT_OPTIONS, defaults);
 
-    // When no formatting options have been specified, default to outputting a
-    // time; e.g.: "9:42 AM".
-    if (Object.keys(filteredOptions).length === 0) {
-        filteredOptions = {
-            hour  : 'numeric',
-            minute: 'numeric',
-        };
+    if (!filteredOptions.hour && !filteredOptions.minute) {
+        // Add default formatting options if hour or minute isn't defined.
+        filteredOptions = Object.assign({hour: 'numeric', minute: 'numeric'}, filteredOptions);
```

Comment:
Use Object rest/spread syntax here `...` instead of depending on `Object.assign` being available in the runtime. The Babel transform will automatically use `Object.assign` of it exists, or an shim helper.